tests/graphql/card.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { Card } from "../../lib/graphql/card";
+
+const cardData = {
+  id: "010e5dcfdd7949fea50a510e97157168",
+  status: "ACTIVE",
+  type: "VISA_BUSINESS_DEBIT",
+  canceledAt: null,
+  expirationDate: "2022-12-18T00:00:00.000Z",
+  orderedAt: "2019-12-18T08:23:52.898Z",
+  holder: "JEAN DUPONT",
+  formattedExpirationDate: "12/22",
+  maskedPan: "6802********5119",
+  settings: {
+    contactlessEnabled: true,
+    cardNotPresentLimits: {
+      daily: {
+        maxAmountCents: 150000,
+        maxTransactions: 25
+      },
+      monthly: {
+        maxAmountCents: 1000000,
+        maxTransactions: 775
+      }
+    },
+    cardPresentLimits: {
+      daily: {
+        maxAmountCents: 450000,
+        maxTransactions: 15
+      },
+      monthly: {
+        maxAmountCents: 2500000,
+        maxTransactions: 465
+      }
+    }
+  }
+};
+
+describe("Card", () => {
+  let sandbox: sinon.SinonSandbox;
+  let client: Client;
+
+  beforeEach(async () => {
+    sandbox = sinon.createSandbox();
+
+    const clientId = "26990216-e340-4f54-b5a5-df9baacc0440";
+    const redirectUri = "https://localhost:3000/auth/callback";
+    const scopes = ["transactions"];
+    const state = "25843739712322056";
+
+    client = new Client({
+      clientId,
+      redirectUri,
+      scopes,
+      state
+    });
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe("#fetch", () => {
+    it("should call rawQuery and return all cards details", async () => {
+      // arrange
+      const cardsData = [
+        cardData,
+        { ...cardData, id: "010e5dcfdd7949fea50a510e97157169" }
+      ];
+      const card = new Card(client.graphQL);
+      const spyOnRawQuery = sandbox.stub(client.graphQL, "rawQuery").resolves({
+        viewer: {
+          mainAccount: {
+            cards: [
+              cardData,
+              { ...cardData, id: "010e5dcfdd7949fea50a510e97157169" }
+            ]
+          }
+        }
+      } as any);
+
+      // act
+      const result = await card.fetch();
+
+      // assert
+      sinon.assert.calledOnce(spyOnRawQuery);
+      expect(result).to.deep.eq({
+        items: cardsData,
+        pageInfo: {
+          hasNextPage: false,
+          hasPreviousPage: false
+        }
+      });
+    });
+  });
+
+  describe("#get", () => {
+    it("should call rawQuery and return card details", async () => {
+      // arrange
+      const card = new Card(client.graphQL);
+      const spyOnRawQuery = sandbox.stub(client.graphQL, "rawQuery").resolves({
+        viewer: {
+          mainAccount: {
+            card: cardData
+          }
+        }
+      } as any);
+
+      // act
+      const result = await card.get({
+        cardId: cardData.id,
+        type: CardType.VisaBusinessDebit
+      });
+
+      // assert
+      sinon.assert.calledOnce(spyOnRawQuery);
+      expect(result).to.deep.eq(cardData);
+    });
+
+    it("should call rawQuery and return null for missing account", async () => {
```

Comment:
`missing card`, right?

Reply:
Hmm it is really testing when account is missing though, e.g.: `resolves({ viewer: {} })`

I wanted to test that sdk is not crashing in that case because it can happen that an sdk user checks cards for an end user which doesn't have account created yet, if it makes sense? 🤔 

If user has missing cards (no cards) gql returns empty array already, so I don't really need to test for that specifically I think

Reply:
got it, you re right :)